### ISSUE TITLE
missing dead code diagnostic

### ISSUE THREAD

**Issue by 91b604ff-5b4e-4821-a182-129f49d1a44c** [llvm/llvm-project#48173]
|  |  |
| --- | --- |
| Bugzilla Link | [48829](https://llvm.org/bz48829) |
| Version | 11.0 |
| OS | Windows NT |
| CC | @zygoloid |

## Extended Description 
for the following code clang will not emit a diagnostic that that f() is never called (dead code). msvc emits the diagnostic correctly.

https://godbolt.org/z/Y5n78v
```
void f();

void _(int num) {
    switch (num)
    {
        case 1:
            break;
            f();
    }
}
```

**Comment by wheatman** [comment]
The code does emit a diagnostic with `-Wunreachable-code` and has since 2.7, but `-Wunreachable-code` is not part of `-Wall`.

I'm closing this since there is a diagnostic, but if there is a belief that this should be moved into `-Wall` it could be opened back up, but I imagine that it might get a little chatty and hence why it is not there now 

the output is 
```
<source>:8:13: warning: code will never be executed [-Wunreachable-code]
    8 |             f();
      |             ^
1 warning generated.
Compiler returned: 0
```
